invenio_files_rest/models.py

```diff
@@ -1156,6 +1233,118 @@ def relink_all(cls, old_file, new_file):
             ObjectVersion.query.filter_by(file_id=str(old_file.id)).update({
                 ObjectVersion.file_id: str(new_file.id)})
 
+    def get_tags(self):
+        """Get tags for object version as dictionary."""
+        return {t.key: t.value for t in self.tags}
+
+    def __eq__(self, other):
+        """Check if the two object are equals."""
+        return other and isinstance(other, self.__class__) and \
+            self.key == other.key and self.file_id == other.file_id
+
+    def __ne__(self, other):
+        """Check if are not equal."""
+        return not self.__eq__(other=other)
+
+    def merge(self, bucket, key=None):
+        """Merge the object inside the bucket."""
+        # check no new version is added while the object is modified
+        orig_ancestor = ObjectVersion.get_versions(
+            bucket=self.bucket, key=key, desc=True)
+        dest_head = ObjectVersion.query_heads_by_bucket(bucket=bucket).filter(
+            ObjectVersion.key == key).one_or_none()
+        if dest_head and all(orig.file_id != dest_head.file_id
```

Comment:
Comment for us when we re-check the code:
This does not ensure that a new version is not added immediately after by another thread or process.
Is it "safe enough" like this, or do we want to introduce an "atomic transaction"?

Reply:
You are right, probably a `with db.session.begin_nested():` wouldn't hurt that much.